taichi/program/async_utils.cpp

```diff
@@ -119,18 +143,33 @@ TaskMeta *get_task_meta(IRBank *ir_bank, const TaskLaunchRecord &t) {
 
   // TODO: this is an abuse since it gathers nothing...
   gather_statements(root_stmt, [&](Stmt *stmt) {
+    if (auto global_load = stmt->cast<GlobalLoadStmt>()) {
```

Comment:
This is the real change. The rest is just to adapt to `snode_or_global_tmp`..

Reply:
Cool -- We may use a snode for global temps, merge `GlobalTemporaryStmt` into `GlobalPtrStmt` and handle this in `get_meta_input_value_states()` (which calls `ControlFlowGraph::gather_loaded_snodes()`) in the future.